app/services/users/archive_by_email.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Users
+  class ArchiveByEmail
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :user
+
+    def call
+      ApplicationRecord.transaction do
+        users_with_matching_email.each do |user_with_matching_email|
+          move_applications(from_user: user_with_matching_email, to_user: user)
+          user.participant_id_changes.create!(from_participant_id: user_with_matching_email.ecf_id, to_participant_id: user.ecf_id) if Feature.ecf_api_disabled?
```

Comment:
```suggestion
           # Move previous change history to new to_user
          from_user.participant_id_changes.update!(user: to_user)
          
          user.participant_id_changes.find_or_create_by!(from_participant_id: user_with_matching_email.ecf_id, to_participant_id: user.ecf_id) if Feature.ecf_api_disabled?
```
just incase we have id changes on old user, also if there are duplicates on the to user